experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/MarquezContext.java

```diff
@@ -1,146 +1,92 @@
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+
+@Slf4j
 public class MarquezContext {
-  private static final Logger logger = LoggerFactory.getLogger(MarquezContext.class);
-  private MarquezWriteOnlyClient marquezClient;
 
-  private String jobNamespace;
-  private String jobName;
-  private String parentRunId;
+  @Getter private static final ObjectMapper mapper = createMapper();
+  @Getter private String jobNamespace;
+  @Getter private String jobName;
+  @Getter private String parentRunId;
+
+  private final CloseableHttpClient httpclient;
 
   private String get(String[] elements, String name, int index) {
     boolean check = elements.length > index + 1 && name.equals(elements[index]);
     if (check) {
       return elements[index + 1];
     } else {
-      logger.warn("missing " + name + " in " + Arrays.toString(elements) + " at " + index);
+      log.warn("missing " + name + " in " + Arrays.toString(elements) + " at " + index);
       return "default";
     }
   }
 
-  public MarquezContext(String argument, Backend marquezBackend) {
-    logger.info("Init MarquezContext: " + argument);
+  public MarquezContext(String argument) {
+    log.info("Init MarquezContext: " + argument);
     String[] elements = argument.split("\\/");
     String version = get(elements, "api", 1);
     if (version.equals("v1")) {
-      logger.info("marquez api v1");
+      log.info("marquez api v1");
     }
     jobNamespace = get(elements, "namespaces", 3);
     jobName = get(elements, "jobs", 5);
     parentRunId = get(elements, "runs", 7);
-    logger.info(String.format("/api/%s/namespaces/%s/jobs/%s/runs/%s", version, jobNamespace, jobName, parentRunId));
-    marquezClient = Clients.newWriteOnlyClient(Backends.newLoggingBackend());
+    log.info(String.format("/api/%s/namespaces/%s/jobs/%s/runs/%s", version, jobNamespace, jobName, parentRunId));
+    httpclient = HttpClients.createDefault();
   }
 
-  public MarquezContext(
-      MarquezWriteOnlyClient marquezClient, String jobNamespace, String jobName, String parentRunId) {
-    super();
-    this.marquezClient = marquezClient;
-    this.jobNamespace = jobNamespace;
-    this.jobName = jobName;
-    this.parentRunId = parentRunId;
-  }
-
-  public DatasetId ensureDatasetExists(String input) {
-    String protocol;
+  public void emit(LineageEvent event) {
     try {
-      URL inputUrl = new URL(input);
-      protocol=inputUrl.getProtocol();
-    } catch (MalformedURLException e) {
-      protocol = "unknown";
+      String json = mapper.writeValueAsString(event);
+      log.info("calling lineage: {}", json);
+
+      HttpPost httpPost = new HttpPost("http://localhost:5000/api/v1/event");
```

Comment:
should these be pulled as env variables?

Reply:
I'd say yes.